File: eMRTDReader/app/src/main/java/com/example/emrtdreader/ResultActivity.java
package com.example.emrtdreader;

import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.emrtdreader.sdk.domain.PassportReadResult;
import com.example.emrtdreader.sdk.models.PassportChipData;
import com.example.emrtdreader.sdk.models.VerificationResult;

public class ResultActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_result);

        PassportReadResult res = (PassportReadResult) getIntent().getSerializableExtra("result");
        ImageView photo = findViewById(R.id.photoImageView);
        TextView data = findViewById(R.id.dataTextView);
        TextView ver = findViewById(R.id.verificationStatusTextView);

        if (res == null) {
            data.setText("No result");
            return;
        }

        PassportChipData d = res.chipData;
        VerificationResult v = res.verification;

        if (d.photoJpeg != null && d.photoJpeg.length > 0) {
            photo.setImageBitmap(BitmapFactory.decodeByteArray(d.photoJpeg, 0, d.photoJpeg.length));
        }

        data.setText(
                "Document: " + d.documentNumber + "\n" +
                "Surname: " + d.surname + "\n" +
                "Given: " + d.givenNames + "\n" +
                "Nationality: " + d.nationality + "\n" +
                "DOB: " + d.dateOfBirth + "\n" +
                "Sex: " + d.sex + "\n" +
                "Expiry: " + d.dateOfExpiry + "\n"
        );

        ver.setText(
                "SOD signature: " + v.sodSignatureValid + "\n" +
                "DG hashes: " + v.dgHashesMatch + "\n" +
                "CSCA trusted: " + v.cscaTrusted + "\n\n" +
                v.details
        );
    }
}

File: eMRTDReader/app/src/main/java/com/example/emrtdreader/ResultActivity.kt
package com.example.emrtdreader

import android.graphics.BitmapFactory
import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.emrtdreader.model.PassportData

class ResultActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_result)

        val photoImageView: ImageView = findViewById(R.id.photoImageView)
        val verificationStatusTextView: TextView = findViewById(R.id.verificationStatusTextView)

        val nameTextView: TextView = findViewById(R.id.nameTextView)
        val passportNumberTextView: TextView = findViewById(R.id.passportNumberTextView)
        val nationalityTextView: TextView = findViewById(R.id.nationalityTextView)
        val dateOfBirthTextView: TextView = findViewById(R.id.dateOfBirthTextView)
        val expiryDateTextView: TextView = findViewById(R.id.expiryDateTextView)
        val personalNumberTextView: TextView = findViewById(R.id.personalNumberTextView)
        val mrzDataTextView: TextView = findViewById(R.id.mrzDataTextView)

        val passportData = intent.getSerializableExtra("PASSPORT_DATA") as? PassportData
        val json = intent.getStringExtra("PASSPORT_JSON").orEmpty()
        val auth = intent.getStringExtra("AUTH_RESULT").orEmpty()

        verificationStatusTextView.text = if (auth.isNotBlank()) "Passive Auth: $auth" else "Passive Auth: N/A"

        if (passportData != null) {
            val fullName = listOf(passportData.surname, passportData.givenNames).filter { it.isNotBlank() }.joinToString(" ")
            nameTextView.text = fullName.ifBlank { "-" }
            passportNumberTextView.text = passportData.documentNumber.ifBlank { "-" }
            nationalityTextView.text = passportData.nationality.ifBlank { "-" }
            dateOfBirthTextView.text = passportData.dateOfBirth.ifBlank { "-" }
            expiryDateTextView.text = passportData.dateOfExpiry.ifBlank { "-" }
            personalNumberTextView.text = passportData.personalNumber.ifBlank { "-" }

            val photo = passportData.photo
            if (photo != null && photo.isNotEmpty()) {
                val bmp = BitmapFactory.decodeByteArray(photo, 0, photo.size)
                if (bmp != null) {
                    photoImageView.setImageBitmap(bmp)
                } else {
                    // Many passports use JPEG2000; BitmapFactory can't decode it.
                    photoImageView.setImageResource(R.drawable.photo_placeholder)
                }
            } else {
                photoImageView.setImageResource(R.drawable.photo_placeholder)
            }
        }

        mrzDataTextView.text = if (json.isNotBlank()) json else "-"
    }
}
